cmake/SetupBOUTThirdParty.cmake

```diff
@@ -19,9 +19,9 @@ if (BOUT_HAS_CUDA)
   set(BOUT_SOURCES_CXX ${BOUT_SOURCES})
   list(FILTER BOUT_SOURCES_CXX INCLUDE REGEX ".*\.cxx")
 
-  set_source_files_properties(${BOUT_SOURCES_CXX} PROPERTIES LANGUAGE CUDA )
+  set_source_files_properties(${BOUT_SOURCES_CXX} PROPERTIES LANGUAGE CUDA)
   find_package(CUDAToolkit)
-  set_target_properties(bout++ PROPERTIES CUDA_STANDARD 14)
+  set_target_properties(bout++ PROPERTIES CUDA_STANDARD 17)
```

Comment:
Is it possible to set this from the value of `CMAKE_CXX_STANDARD`?

Reply:
No, because we don't set `CMAKE_CXX_STANDARD` but instead use `target_compile_features` on the bout++ target. However, the point of avoiding duplication is a good one. We can remove setting the `CUDA_STANDARD` for CUDA compilation since `target_compile_features` for `cxx_std_17` implicitly enables `cuda_std_17`. How does that sound?

Reply:
Maybe a comment in the code that `CUDA_STANDARD` is implicitly set, so someone doesn't re-add it?